airflow/models/dagrun.py

```diff
@@ -326,7 +326,7 @@ def get_task_instances(self, state=None, session=None) -> Iterable[TI]:
 
         if self.dag and self.dag.partial:
             tis = tis.filter(TI.task_id.in_(self.dag.task_ids))
-        return tis.all()
+        return tis.order_by(TI.execution_date).all()
```

Comment:
This does not make sense, L308 has an explicit filter to get TIs with a single execution_date (for this DagRun).

So you will only ever have a single execution_date for all the TIs returned by this query, so no point in ordering.

https://github.com/apache/airflow/blob/d1d02b62e3436dedfe9a2b80cd1e61954639ca4d/airflow/models/dagrun.py#L308-L311